### PR TITLE
Add lobby filtering

### LINKED ISSUE #21: Lobby filtering
Add some basic filters for lobby:
1. hide full rooms
2. hide rooms with started game

That's really all we need now 😃

### PR DESCRIPTION
**Pull request by pastpatryk**
Adds ability to filter matches by custom name or game type

image

Closes #21

### CODE REVIEW

**Comment by glowacki-dev** [comment]
image
This box has a small margin on top which obstructs the bottom of search bar and captures click events - if you want to double-click on a search bar to select all of it's content you have to do it on a top part, otherwise only the first click gets registered

File: src/components/lobby/FilterBox.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        ref={searchRef}
+        placeholder="Search..."
+        onFocus={() => setTimeout(() => setOpen(true), 300)}
+        value={filters.query}
+        onChange={handleQueryChange}
+        fluid
+      />
+      <div className={styles.filterStatus}>
+        {selectedGames.length === games.length ? (
+          <span className={styles.statusLabel}>All games</span>
```
Comment (glowacki-dev) [inline src/components/lobby/FilterBox.js:77]: This should be moved to locales

File: src/components/lobby/FilterBox.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    const searchInput = searchRef?.current?.inputRef?.current;
+    if (!searchInput || e.target !== searchInput) {
+      setOpen(false);
+    }
+  };
+
+  const handleToggleGame = (game) => {
+    let { games } = filters;
+    if (selectedGames.includes(game)) {
+      if (selectedGames.length === 1) return;
```
Comment (glowacki-dev) [inline src/components/lobby/FilterBox.js:49]: Alternatively, deselecting all games could disable this filter = automatically select all games

**Comment by pastpatryk** [comment]
> image
> This box has a small margin on top which obstructs the bottom of search bar and captures click events - if you want to double-click on a search bar to select all of it's content you have to do it on a top part, otherwise only the first click gets registered

Interesting... it looks different on my browser 🤔

**Comment by glowacki-dev** [comment]
> 
> 
> > image
> > This box has a small margin on top which obstructs the bottom of search bar and captures click events - if you want to double-click on a search bar to select all of it's content you have to do it on a top part, otherwise only the first click gets registered
> 
> Interesting... it looks different on my browser 🤔

I've added the purple border to highlight the area occupied by this box 🙈 I've checked in two browsers and both of them have this transparent top margin that goes over the search bar

File: src/components/lobby/FilterBox.js (hunk 2 above)
Comment (pastpatryk) [reply src/components/lobby/FilterBox.js]: done 👍

**Comment by pastpatryk** [comment]
> > > image
> > > This box has a small margin on top which obstructs the bottom of search bar and captures click events - if you want to double-click on a search bar to select all of it's content you have to do it on a top part, otherwise only the first click gets registered
> > 
> > 
> > Interesting... it looks different on my browser 🤔
> 
> I've added the purple border to highlight the area occupied by this box 🙈 I've checked in two browsers and both of them have this transparent top margin that goes over the search bar

@glowacki-dev fixed 👍

**Review by glowacki-dev — APPROVED** [review]

**Pull request merged**